template/getter.go

```diff
@@ -78,9 +90,34 @@ func (t templatedConfigGetter) Get(configID string) (*api.Config, error) {
 	return &configCopy, nil
 }
 
+func (t templatedConfigGetter) GetAndFlagSecretData(configID string) (*api.Config, bool, error) {
```

Comment:
Non-blocking:  Would it make sense to basically replace the `Get` implementation with a call to `GetAndFlagSecretData`, and just leave out the flag in the return value?  Since they otherwise seem to be exactly the same function.